irohad/ametsuchi/impl/flat_file/flat_file.cpp

```diff
@@ -21,96 +21,80 @@
+        std::vector<boost::filesystem::path> ps;
+        std::copy(boost::filesystem::directory_iterator{dump_dir},
+                  boost::filesystem::directory_iterator{},
+                  std::back_inserter(ps));
+        std::sort(ps.begin(),
+                  ps.end(),
+                  [](const boost::filesystem::path &lhs,
+                     const boost::filesystem::path &rhs) {
+                    return lhs.compare(rhs) < 0;
+                  });
+        return ps;
+      }();
+
+      auto const missing = boost::range::find_if(
+          files | boost::adaptors::indexed(1), [](const auto &it) {
+            return id_to_name(it.index()) != it.value().filename();
+          });
+
+      std::for_each(
+          missing.get(), files.cend(), [](const boost::filesystem::path &p) {
+            boost::filesystem::remove(p);
+          });
+
+      return missing.get() - files.cbegin();
     }
+  }  // namespace ametsuchi
+}  // namespace iroha
 
-    auto const files = [&dump_dir] {
-      std::vector<boost::filesystem::path> ps;
-      std::copy(boost::filesystem::directory_iterator{dump_dir},
-                boost::filesystem::directory_iterator{},
-                std::back_inserter(ps));
-      std::sort(ps.begin(),
-                ps.end(),
-                [](const boost::filesystem::path &lhs,
-                   const boost::filesystem::path &rhs) {
-                  return lhs.compare(rhs) < 0;
-                });
-      return ps;
-    }();
-
-    auto const missing = boost::range::find_if(
-        files | boost::adaptors::indexed(1), [](const auto &it) {
-          return id_to_name(it.index()) != it.value().filename();
-        });
-
-    std::for_each(
-        missing.get(), files.cend(), [](const boost::filesystem::path &p) {
-          boost::filesystem::remove(p);
-        });
-
-    return missing.get() - files.cbegin();
-  }
-}  // namespace
+using namespace iroha::ametsuchi;
 
 // ----------| public API |----------
 
 std::unique_ptr<FlatFile> FlatFile::create(const std::string &path) {
   auto log_ = logger::log("FlatFile::create()");
 
-  if (boost::filesystem::create_directory(path)) {
-    if (!boost::filesystem::is_directory(path)) {
-      log_->error("Cannot create storage dir: {}", path);
-    }
-  }
-  auto res = check_consistency(path);
-  if (not res) {
-    log_->error("Checking consistency for {} - failed", path);
+  boost::system::error_code err;
+  if (not boost::filesystem::is_directory(path, err)
+      and not boost::filesystem::create_directory(path, err)) {
+    log_->error("Cannot create storage dir: {}\n{}", path, err.message());
     return nullptr;
   }
-  return std::unique_ptr<FlatFile>(new FlatFile(*res, path));
+
+  auto res = check_consistency(path);
+  return std::make_unique<FlatFile>(*res, path, private_tag{});
 }
 
-void FlatFile::add(Identifier id, const std::vector<uint8_t> &block) {
+bool FlatFile::add(Identifier id, const std::vector<uint8_t> &block) {
```

Comment:
In nearest future, we want to rework interfaces of storage with `Result` type. So, I request to write todo with reworking method with respect to this behavior.